tsa/scrape_pdf/model.py

```diff
@@ -0,0 +1,47 @@
+import vertexai
+from vertexai.generative_models import GenerativeModel
+from vertexai.preview import generative_models
+
+MODEL = "gemini-1.5-pro-preview-0409"
+PROJECT = ""
+REGION = "us-central1"
+
+
+def generate(text, generation_config, safety_settings):
+    vertexai.init(project=PROJECT, location=REGION)
+    model = GenerativeModel(MODEL)
+    response = model.generate_content(
+        contents=[text],
+        generation_config=generation_config,
+        safety_settings=safety_settings,
+        stream=False,
+    )
+    return response.text.strip(" \n")
+
+
+def extract_date(filename):
+    text = f"""
+    you are an assistant and assigned with the task of extracting dates from an input string
+
+    input string: \"{filename}\",
+
+    return the dates using the format YYYY-MM-DD, YYYY-MM-DD
```

Comment:
yeah as you said, might just use a regex here and save llm for more potent stuff?

Reply:
yeah, pretty certain will replace with regex. Planning to come back to this.